test/cloudwatchlogs/publish_logs_test.go

```diff
@@ -113,6 +113,35 @@ func TestWriteLogsToCloudWatch(t *testing.T) {
 	}
 }
 
+// TestAutoRemovalStopAgent configures agent to monitor a file with auto removal on.
+// Then it restarts the agent.
+// Verify the file is NOT removed.
+func TestAutoRemovalStopAgent(t *testing.T) {
+	// Use instance id so 2 tests in parallel on different machines do not conflict.
+	instanceId := awsservice.GetInstanceId()
+	defer awsservice.DeleteLogGroupAndStream(instanceId, instanceId)
+	fpath := logFilePath + "1"
+	f, err := os.Create(fpath)
+	if err != nil {
+		t.Fatalf("Error occurred creating log file for writing: %v", err)
+	}
+	defer f.Close()
+	defer os.Remove(fpath)
+	configPath := "resources/config_auto_removal.json"
+	common.StartAgent(configPath, true, false)
+	// Sleep 20 seconds before and after writing the file to ensure the agent reads it.
+	sleepTime := time.Second*20
+	time.Sleep(sleepTime)
+	writeLogs(t, f, 1000)
+	time.Sleep(sleepTime)
+	common.StopAgent()
+	time.Sleep(sleepTime)
+	assert.FileExists(t, fpath, "file does not exist, {}", fpath)
+	common.StartAgent(configPath, true, false)
```

Comment:
Should you stop the agent again after this one, or does the test framework handle that?

Reply:
No the framework does not.
You are right, safer to stop it.
No reason to leave it running.
I was lazy and left it running figuring it doesn't really matter.